### DISCUSSION TITLE
VPN Disconnected after losing the signal

### DISCUSSION THREAD

**Discussion by maaaom** [strongswan/strongswan#2239] [Q&A]
Hi,

I'm experiencing an issue with the Android app (version 2.5.1) regarding network reconnection. When I enter an area without mobile coverage and subsequently lose data connectivity, the app still displays as "Connected" upon regaining coverage. However, I'm unable to access any services unless I manually **disconnect and reconnect.**

Could you please implement an automatic reconnection feature to address this issue when the app detects that signal has been restored?

Thank you for your assistance!

Best regards,

**Comment by tobiasbrunner** [comment]
That's probably because your server kills the SA too early (e.g. because it has relatively aggressive DPD/retransmission settings). You might want to change the configuration there.

**Reply by maaaom** [reply]
I configure DPDdelay: 5m

What should be ?

**Reply by tobiasbrunner** [reply]
I'd set it to > 1h (or completely disable it so only possible rekeyings trigger an implicit DPD, but I'd also increase the [lifetimes](https://docs.strongswan.org/docs/5.9/config/rekeying.html) there) to only remove completely abandoned SAs (if you use uniqueness checks, old SAs will be removed anyway when a client connects from scratch).

**Reply by maaaom** [reply]
Thank you for the clarification. If I set the DPD interval to 2 hours, and I am out of coverage for more than 2 hours, will I encounter the same connection issue upon returning to a coverage area?

Additionally, my current IKE lifetime is set to 24 hours, and the key lifetime is 8 hours. If I adjust the DPD to 24 hours, considering that the IKE lifetime is also 24 hours, could this resolve the issue?

**Reply by tobiasbrunner** [reply]
> If I set the DPD interval to 2 hours, and I am out of coverage for more than 2 hours, will I encounter the same connection issue upon returning to a coverage area?

No, you shouldn't. As long as the server still has the SAs installed, the client can continue to use them even if it was unreachable for a while.

> Additionally, my current IKE lifetime is set to 24 hours, and the key lifetime is 8 hours. If I adjust the DPD to 24 hours, considering that the IKE lifetime is also 24 hours, could this resolve the issue?

You don't need DPD then as every IKE exchange (i.e. rekeyings as well) will act as DPD. Note that the client will rekey child SAs every 2 hours (or at least attempt to do so).

**Reply by maaaom** [reply]
So by your experience,  I need the best practice that should I do? to avoid any issue even If I will be long time out of the coverage !

Also please share me the steps how I can do it.

Thanks in advance for your help.

**Reply by tobiasbrunner** [reply]
I'd just disable active DPDs (as I said, the rekeying exchanges will acts as DPDs as well on abandoned SAs). Unless you need clients to be able to connect from multiple devices at the same time, then I'd also enable the uniqueness checks, so abandoned SAs get deleted when a client reconnects from scratch (can happen if it is unable to rekey or terminate the SA because there is no connectivity at that time).
